sklearn/linear_model/tests/test_least_angle.py

```diff
@@ -730,3 +730,36 @@ def test_lasso_lars_fit_copyX_behaviour(copy_X):
     y = X[:, 2]
     lasso_lars.fit(X, y, copy_X=copy_X)
     assert copy_X == np.array_equal(X, X_copy)
+
+
+@pytest.mark.parametrize('y_list, expected_y', [
+    ([-2.5, -2.5], [0, 2.5, 0, 2.5, 0]),
+    ([[-2.5, -2.5], [-2.5, -2.5]],
+     [[0, 5, 0, 2.5, 0], [0, 5, 0, 2.5, 0]])])
+def test_lars_with_jitter(y_list, expected_y):
+    """
+    Test that user input of a small amount of jitter,
+    using example provided in issue #2746
+
+    """
+
+    X = np.array([[0.0, 0.0, 0.0, -1.0, 0.0], [0.0, -1.0, 0.0, 0.0, 0.0]])
+    y = np.array(y_list)
+    expected_output = np.array(expected_y)
+    alpha = 0.001
+    fit_intercept = False
```

Comment:
why forcing fit_intercept = False?

Reply:
This is the at the edge of my stats/linear algebra understanding, but I _think_ we need to force it to be `False` since the error only occurs for exactly aligned values (e.g. [this comment](https://github.com/scikit-learn/scikit-learn/issues/2746#issue-25521224)).

Reply:
from your comment I understand that the test would not be a non-regression test with fit_intercept=True as you only see an error with fit_intercept. However below the X and y you chose do work too with jitter=None. Did you check that the test still pass with fit_intercept=True?